enterprise/cmd/frontend/internal/auth/openidconnect/config.go

```diff
@@ -6,49 +6,54 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
-	"net/http"
 
-	"github.com/inconshreveable/log15"
 	"github.com/sourcegraph/log"
 
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/auth/providers"
 	"github.com/sourcegraph/sourcegraph/enterprise/internal/licensing"
 	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
 	"github.com/sourcegraph/sourcegraph/schema"
 )
 
-var mockGetProviderValue *provider
+var mockGetProviderValue *Provider
 
-// getProvider looks up the registered openidconnect auth provider with the given ID.
-func getProvider(id string) *provider {
+// getProvider looks up the registered OpenID Connect authentication provider
+// with the given ID.
+func getProvider(id string) *Provider {
 	if mockGetProviderValue != nil {
 		return mockGetProviderValue
 	}
-	p, _ := providers.GetProviderByConfigID(providers.ConfigID{Type: providerType, ID: id}).(*provider)
+	p, _ := providers.GetProviderByConfigID(
+		providers.ConfigID{
+			Type: providerType,
+			ID:   id,
+		},
+	).(*Provider)
 	return p
```

Comment:
I'm not sure if I read the code right, but it seems to me that `GetProviderByConfigID` returns only 1 value of a type [`Provider`](https://github.com/sourcegraph/sourcegraph/blob/cce473f9e9cfa8b0580440c8a07bfe152d361bba/cmd/frontend/auth/providers/providers.go#L181:6) - based on precise code intel.

1. Why are we assigning 2 variables here?
2. Can we get rid of the casting and use the value directly?

Based on my completely ignorant golang expertise, this should be enough?
```suggestion
	p := providers.GetProviderByConfigID(
		providers.ConfigID{
			Type: providerType,
			ID:   id,
		},
	)
	return &p
```

Reply:
🤔 `providers.GetProviderByConfigID` [returns an interface `providers.Provider`](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@c34d4dda35447bc83e485cc3f9328b2373d14050/-/blob/cmd/frontend/auth/providers/providers.go?L181:6-181:27), I think that's where the casting is coming from.

But I think you're right about we don't need to assign 2 variables, given we're doing a force-casting anyway.

Reply:
Actually... realized we were using the ok-pattern for type assertion, which in case the `providers.GetProviderByConfigID` returns `nil` (i.e. `.(*Provider)` fails) and the code won't panic.